compiler/rustc_mir_dataflow/src/framework/engine.rs

```diff
@@ -99,13 +107,18 @@ where
     T: Idx,
 {
     /// Creates a new `Engine` to solve a gen-kill dataflow problem.
-    pub fn new_gen_kill(tcx: TyCtxt<'tcx>, body: &'a mir::Body<'tcx>, analysis: A) -> Self {
+    pub fn new_gen_kill(
+        tcx: TyCtxt<'tcx>,
+        body: &'a mir::Body<'tcx>,
+        analysis: A,
+        blocks: Blocks,
```

Comment:
Why did we have to add this flag? I suspect this is related to the unwinding comments that I added above, though I'm not sure.

Reply:
It's related to https://github.com/rust-lang/rust/pull/104488#issuecomment-1365802291. 
In the fixpoint alg we push all mir blocks in the work queue, even if they are not reachable from the start for a forward analysis, essentially negating the effects of the `dead_unwind` flag in some cases. This is undesirable for drop elaboration due to the desugaring of `DropAndReplace` so I added a flag to control whether we initially insert all blocks in the work queue. 
All other analyses have the same behavior as before but maybe some could benefit from this optimization as well.

Reply:
I'm confused -- why does the desugaring of drop-and-replace have an effect here? The only reason I can think of that it would generate a lot of dead code would be if you had panic=abortmode?

Reply:
Consider something like 
```rust
fn single() {
    let x;
    noop();
    x = Big::new();
}
```
The `maybe_init` analisys on the MIR right before drop elaboration will look like (without any modifications to the dataflow engine): 
<img src="https://user-images.githubusercontent.com/21265557/209887580-39880211-9157-432a-8c3b-d537ad81948e.svg" width="600" />

The assignment `x = Big::new()` is now desugared into `drop(_1)` in bb2 and  `_1 = move _3` in bb3 and bb4.
Now, that `drop(_1)` in bb2 is a noop since `_1` was not assigned before, and bb4 is unreachable, but drop elaboration hasn't run and removed it yet. 
`drop(_3)` in bb3 is also a noop because we just moved out of `_3`.
However, by pushing all blocks in the dataflow engine work queue, bb4 might propagate its effects to its successors, in this case the variable `_1` being live, which might pollute other legit unwind blocks like bb7.
What we end up with is incorrect init data in bb7, since at that point in the CFG we can't possible have `_1` live.

If instead we only push in the queue blocks that are actually reachable from the start bb4 will not be considered and will not influence other blocks.

Reply:
Anyway, I think we can remove the flag and let this be the default behavior for forward analyses. I think it's already what the user mean by providing `dead_unwinds`

Reply:
should be fixed by https://github.com/rust-lang/rust/pull/106430